src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -246,10 +251,15 @@ public void OnTimeout(TimeoutReason reason)
                 case TimeoutReason.TimeoutFeature:
                     SendGoAway(GetHighestStreamId()).Preserve();
                     break;
-                case TimeoutReason.RequestHeaders: // Request header timeout is handled in starting stream queue
-                case TimeoutReason.KeepAlive:  // Keep-alive is handled by msquic
                 case TimeoutReason.ReadDataRate:
+                    Abort(new ConnectionAbortedException(CoreStrings.BadRequest_RequestBodyTimeout), Http3ErrorCode.InternalError);
+                    break;
                 case TimeoutReason.WriteDataRate:
+                    Log.ResponseMinimumDataRateNotSatisfied(_context.ConnectionId, traceIdentifier: null);
+                    Abort(new ConnectionAbortedException(CoreStrings.ConnectionTimedBecauseResponseMininumDataRateNotSatisfied), Http3ErrorCode.InternalError);
+                    break;
```

Comment:
This seems about the same as HttpConnection.OnTimeout(). Do you think it's worth changing Http3ConnectionMiddleware to call through HttpConnection like HttpConnectionMiddleware does?

Then we could make Http3Connection an IRequestProcessor like Http1Connection and Http2Connection giving us greater consistency. HttpConnection itself is pretty small, so I think it be a fairly small change.

Reply:
Discussed offline. Will attempt in a follow-up PR.